### PR TITLE
get rid of stupid deprecation warning

### PR DESCRIPTION
**Pull request by MrIvanPlays**
gets rid of this crap
image

### CODE REVIEW

**Review by Jikoo — APPROVED** [review]
👍

**Comment by MrNemo64** [comment]
Wouldn't this block all deprecated event messages? Can we make the filter more precise?

File: platform/bukkit/plugin/src/main/java/me/lokka30/treasury/plugin/bukkit/TreasuryBukkit.java [hunk 1]
```diff
@@ -50,6 +51,16 @@
      */
     @Override
     public void onEnable() {
+        // get rid of this warning: https://img.mrivanplays.com/jaqzdjmvpz.png .We register a
+        // listener for our events for backwards compatibility. Couldn't bukkit be smarter and not
+        // print it for events which are created by the plugin which is listening??
```
Comment (lokka30) [inline platform/bukkit/plugin/src/main/java/me/lokka30/treasury/plugin/bukkit/TreasuryBukkit.java:56]: should I submit an issue to Paper about this?

**Review by lokka30 — APPROVED** [review]

**Comment by yannicklamprecht** [comment]
```java
    private static final List<Class<?>> ownDeprecatedClasses = Arrays.asList(
            me.lokka30.treasury.api.economy.event.AccountEvent.class,
            me.lokka30.treasury.api.economy.event.AccountTransactionEvent.class,
            me.lokka30.treasury.api.economy.event.NonPlayerAccountTransactionEvent.class,
            me.lokka30.treasury.api.economy.event.PlayerAccountTransactionEvent.class
    );
  ``



```java
getLogger().setFilter(record -> {
            if (ownDeprecatedClasses.stream().anyMatch(aClass -> record.getMessage().contains(aClass.getName()))
                    && record.getMessage().contains("but the event" +
                    " is Deprecated")) {
                return false;
            }
            return oldFilter == null || oldFilter.isLoggable(record);
        });
```

Should do it. Probably

**Review by yannicklamprecht — CHANGES_REQUESTED** [review]
```java
    private static final List<Class<?>> ownDeprecatedClasses = Arrays.asList(
            me.lokka30.treasury.api.economy.event.AccountEvent.class,
            me.lokka30.treasury.api.economy.event.AccountTransactionEvent.class,
            me.lokka30.treasury.api.economy.event.NonPlayerAccountTransactionEvent.class,
            me.lokka30.treasury.api.economy.event.PlayerAccountTransactionEvent.class
    );
  ```



```java
getLogger().setFilter(record -> {
            if (ownDeprecatedClasses.stream().anyMatch(aClass -> record.getMessage().contains(aClass.getName()))
                    && record.getMessage().contains("but the event" +
                    " is Deprecated")) {
                return false;
            }
            return oldFilter == null || oldFilter.isLoggable(record);
        });
```

Should do it. Probably

**Comment by MrNemo64** [comment]
I do like that filter more

**Comment by yannicklamprecht** [comment]
> I do like that filter more. Since the filter will probably run pretty often maybe we should consider more optimal ways of checking?

Logging is done on a separate thread but looking into converting that stream into a "primitive" loop would be a good idea.
First checking for  `but the event is Deprecated` and then looping through the deprecated classes should be better.
Additionally we can just initialize the list with the name of that class. Will comment it in a few.

File: platform/bukkit/plugin/src/main/java/me/lokka30/treasury/plugin/bukkit/TreasuryBukkit.java (hunk 1 above)
Comment (yannicklamprecht) [reply platform/bukkit/plugin/src/main/java/me/lokka30/treasury/plugin/bukkit/TreasuryBukkit.java:56]: I'm a fan of adding a class annotation for that.

From that example a missing annotation should not throw a CNFE.
https://www.baeldung.com/classnotfoundexception-missing-annotation

So something like that could be possible in Paper-API:

```java
@Retention(RetentionPolicy.RUNTIME)
    @Target(ElementType.TYPE)
    public @interface DoNotWarnFor {
        Class<?>[] plugins();
    }
    
    @Deprecated
    @DoNotWarnFor(plugins = TreasuryBukkit.class)
    public class SomeEvent extends Event {
        
        @Override
        public @NotNull HandlerList getHandlers() {
            return null;
        }
    }
```

That could be filtered.

Comment (yannicklamprecht) [reply platform/bukkit/plugin/src/main/java/me/lokka30/treasury/plugin/bukkit/TreasuryBukkit.java:56]: But that is the lazy variant, better would be to compare the class loader of the event class with the class loader of there registering class.

**Comment by lokka30** [comment]
Thanks @yannicklamprecht, that solution is much better.

**Comment by MrIvanPlays** [comment]
I will convert the pr with yannick's solution and merge.

File: platform/bukkit/plugin/src/main/java/me/lokka30/treasury/plugin/bukkit/TreasuryBukkit.java (hunk 1 above)
Comment (MrIvanPlays) [reply platform/bukkit/plugin/src/main/java/me/lokka30/treasury/plugin/bukkit/TreasuryBukkit.java:56]: Thats just a comment explaining why is that code. What the fuck have you actually did here?

**Comment by MrNemo64** [comment]
I just thought of something, will this block all deprecation warning messages from all plugins or only from treasury?

**Comment by MrIvanPlays** [comment]
> I just thought of something, will this block all deprecation warning messages from all plugins or only from treasury?

treasury only

**Comment by lokka30** [comment]
> I just thought of something, will this block all deprecation warning messages from all plugins or only from treasury?

it blocks from Treasury's logger :) thru `Plugin#getLogger()`

**Pull request merged**